### PR TITLE
Removes deleted site from displaying

### LINKED ISSUE #15592: If you delete your last site it stays visible until restart
### Expected behavior
If I delete my only site I should be presented with the 'Add a site' empty view again

### Actual behavior
The last site stays visible and I am able to attempt to interact with it

### Steps to reproduce the behavior

1. Sign in with an account with no sites
2. Tap the add site button
3. Follow the steps to create a site
4. Once its created go to Site Settings
5. Tap the Delete Site button
6. Enter your site name to delete the site

##### Tested on [device], Android [version], WPAndroid [version]
Android Simulator, WPAndroid 18.6

### PR DESCRIPTION
**Pull request by SowmiaS**
Fixes #[15592](https://github.com/wordpress-mobile/WordPress-Android/issues/15592)

## Now
Once the site is deleted, it should be reflected immediately in the app.
Now, the deleted site is still displayed until the app restarts.

## Description
So this PR fixes this issue by reflecting the deleted site immediately in the app.

### Before 
https://user-images.githubusercontent.com/4975407/157210184-205888f5-0f4d-49a8-b35c-e5d579f42d4a.mp4
### After
https://user-images.githubusercontent.com/4975407/157209770-e8ba4565-2b05-4579-8c78-7b2daf5ea237.mp4

#### Test 1
1. when there are **more than two sites**, delete a site.
2. Then it should delete the site
3. And Then navigate to the `Choose site` screen.
#### Test 2
1. when there are **two sites**, delete a site.
2. Then it should delete the site
3. And Then navigate to the `My Site` screen with the remaining site being selected automatically.
#### Test 3
1. when there are **one sites**, delete a site.
2. Then it should delete the site
3. And Then navigate to the `Create a new site` screen with the remaining site being selected automatically.

#### Test 3
1. when the user is neither signed in to a WordPress.com account nor if he has a .org site.
2. And deletes the site
4. And Then navigate to the `Signin` screen.

## Regression Notes
1. No Potential unintended areas of impact.
2. Adds unit test for the changes in WPMainActivityViewModel where possible.

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml
Comment (AjeshRPai) [inline .idea/inspectionProfiles/Project_Default.xml]: @SowmiaS Please add these files to your git ignore local file. We can exclude these changes from the PR as these are only required for your local env

File: WordPress/src/main/java/org/wordpress/android/extension/AndroidExtensions.kt
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/extension/AndroidExtensions.kt]: @SowmiaS 👍 for extracting the `ActivityLauncher.showSignInForResultJetpackOnly` and `ActivityLauncher.showSignInForResult` to a method. 🙌 .I think we can move this function to the `WPMainActivity`. I think that `showSignInForResult` is not an extension required for `FragmentActivity`. We can move this as a normal function to the `WPMainActivity`. Wdyt?

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java]: This `return` statement is redundant as this is the last statement.

File: WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt]: @SowmiaS 👍  on adding these dependencies to the `WPMainActivityViewModel` from the Activity. We can rename these variables without the 'm' prefix as that was a convention followed in the java code. for eg: `mSiteStore` can be renamed to `siteStore`. wdyt? 🤔

File: WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt
Comment (AjeshRPai) [inline WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt]: @SowmiaS  We can rename these variables without the 'm' prefix as well. 🤔

File: WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt]: Can be renamed to `hasMultipleSites`. Wdyt?. 🤔

File: WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt
Comment (AjeshRPai) [inline WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt:692]: @SowmiaS I think an import was missing in this file for the `mock` helper function, Please make sure to push it as well. 
import statement - `import com.nhaarman.mockitokotlin2.mock`

File: WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt
Comment (AjeshRPai) [inline WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt:115]: As the site returned by `firstSite` variable is the one selected by default in the site picker screen, can we rename this to `selectedSite`?. WDYT? 🤔

File: WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt
Comment (AjeshRPai) [inline WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt]: We can remove the `be` from the test statements. 

`firstSite should be return the first site available in the list of sites` can be changed to 
`selectedSite should return the first site available in the list of sites`

**Review by AjeshRPai — CHANGES_REQUESTED** [review]
@SowmiaS Great work. 🙌 

I have tested the changes and everything looks good to me. I have tested for all the scenarios described in the parent issue and in the test block of the PR description. Everything works as Intended. I have left some minor comments and suggestions. 
Hope it helps, Let me know if anything is not clear.

File: WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt
Comment (SowmiaS) [reply WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt:115]: At this point of time, there are actually no selected sites. So naming the property as first site makes it clear that we are sending first site, which will be considered as selected site in site picker activity.
```kotlin
ActivityLauncher.showSitePickerForResult(this, mViewModel.getFirstSite()); // It seems clear that we are sending first site as selected site
ActivityLauncher.showSitePickerForResult(this, mViewModel.getSelectedSite());
```
@AjeshRPai If you still feel, it makes more sense when it is changed as `selectedSite`, I can do it. WDYT? 🤔

File: WordPress/src/main/java/org/wordpress/android/extension/AndroidExtensions.kt
Comment (SowmiaS) [reply WordPress/src/main/java/org/wordpress/android/extension/AndroidExtensions.kt]: @AjeshRPai It makes sense 🙂 . I have moved the function to `ActivityLauncher`, so that related code will be in one class. Done in commits - https://github.com/wordpress-mobile/WordPress-Android/pull/16063/commits/352e063c62c610fa98aa17dd2e117f981356fb75, https://github.com/wordpress-mobile/WordPress-Android/pull/16063/commits/47477c0359b67045f7701425132afe039ad94e40

**Comment by SowmiaS** [comment]
Thanks for taking your time to review this PR @AjeshRPai 🙇 . 

I agree with all the review comments and I have fixed all the comments given by you except **renaming `firstSite` to `selectedSite`**. 

> At this point of time, there are actually no selected sites. So naming the property as first site makes it clear that we are sending first site, which will be considered as selected site in site picker activity.
> 
```kotlin 
ActivityLauncher.showSitePickerForResult(this, mViewModel.getFirstSite()); // It seems clear that we are sending first site as selected site
ActivityLauncher.showSitePickerForResult(this, mViewModel.getSelectedSite());
```

@AjeshRPai If you still feel, it makes more sense when it is changed as `selectedSite`, I can do it. WDYT? 🤔

Kindly have a look at my changes.

File: WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt
Comment (SowmiaS) [reply WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt]: I agree @AjeshRPai. Done in commit https://github.com/wordpress-mobile/WordPress-Android/pull/16063/commits/ec02f37cda30b9213a1bfb3c563fff4f0869a0c5.

File: WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt
Comment (SowmiaS) [reply WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt]: I agree @AjeshRPai. Done in commit https://github.com/wordpress-mobile/WordPress-Android/pull/16063/commits/ec02f37cda30b9213a1bfb3c563fff4f0869a0c5.

File: WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt
Comment (SowmiaS) [reply WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt]: It makes sense @AjeshRPai 🙂 . I have renamed to `hasMultipleSites`. Done in commit https://github.com/wordpress-mobile/WordPress-Android/pull/16063/commits/8bcb6feea611ad7bc7154d9dd89b72c73efe2c3b.

File: WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt
Comment (SowmiaS) [reply WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt:692]: @AjeshRPai. I have added the missing import in commit https://github.com/wordpress-mobile/WordPress-Android/pull/16063/commits/0b8d8798b81324cf3ef9972c49c7c9ae904745ef

File: .idea/inspectionProfiles/Project_Default.xml
Comment (SowmiaS) [reply .idea/inspectionProfiles/Project_Default.xml]: This was committed by mistake. Reverted back the changes done in inspectionprofiles. Done in commit https://github.com/wordpress-mobile/WordPress-Android/pull/16063/commits/f0de81d79fc75efdd570528ae43bf8fe94935edc

File: WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
Comment (SowmiaS) [reply WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java]: I have removed the redundant return statement in [commit](https://github.com/wordpress-mobile/WordPress-Android/pull/16063/commits/643b8dec6a76175a2964cdfdc6294e0ec9551aa3).

File: WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt
Comment (SowmiaS) [reply WordPress/src/test/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModelTest.kt]: I have corrected the test function name as below
- firstSite should return null when there are no sites
- firstSite should return the first site available in the list of sites
 Done in commit https://github.com/wordpress-mobile/WordPress-Android/pull/16063/commits/03949a4d8ee0977530001486f2e4ae43a49c6c2c.

**Comment by AjeshRPai** [comment]
@SowmiaS, Thanks for making the changes 👍 . I am working on some other tasks which have a higher priority at the moment. I will take a look at these changes/ or assign someone as soon as possible. In the meanwhile may I suggest that you   update the PR by merging the [wordpress-mobile:trunk](https://github.com/wordpress-mobile/WordPress-Android)

**Comment by AliSoftware** [comment]
Note: don't forget to add a line to mention your fix in `RELEASE-NOTES.txt` before it gets merged 🙂

File: .idea/inspectionProfiles/Project_Default.xml
Comment (AjeshRPai) [reply .idea/inspectionProfiles/Project_Default.xml]: @SowmiaS I think there is still some change in the file which is showing up in the  `Files changed` tab of the PR. Please have a look at it.

File: WordPress/src/main/java/org/wordpress/android/extension/AndroidExtensions.kt
Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/extension/AndroidExtensions.kt]: Suggestion 💡   @SowmiaS I think its better to have the method `showSignInForResultBasedOnIsJetpackAppBuildConfig` in `WPMainActivity` as this is only being used in the `WPMainActivity`. In addition to that, I feel the `ActivityLauncher` class should be responsible only to launch activity and not hold any logic. WDYT?. 🤔

File: WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt
Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt:115]: >At this point of time, there are actually no selected sites. So naming the property as first site makes it clear that we are sending first site, which will be considered as selected site in site picker activity.

@SowmiaS 👋 I think you are right. Let's keep the naming as `firstSite` itself. 🙌

**Comment by AjeshRPai** [comment]
Thanks, @AliSoftware for pointing it out. 🙌 

@SowmiaS Thanks for making the changes. I have added a minor comment and a suggestion. Please have a look at it. 

As @AliSoftware Pointed out, Could you add a line to mention your fix in ` RELEASE-NOTES.txt` as well?. You can find some examples in the same file.

**Review by AjeshRPai — CHANGES_REQUESTED** [review]
Thanks, @AliSoftware for pointing it out. 🙌 

@SowmiaS Thanks for making the changes. I have added a minor comment and a suggestion. Please have a look at it. 

As @AliSoftware Pointed out, Could you add a line to mention your fix in ` RELEASE-NOTES.txt` as well?. You can find some examples in the same file.

File: WordPress/src/main/java/org/wordpress/android/extension/AndroidExtensions.kt
Comment (SowmiaS) [reply WordPress/src/main/java/org/wordpress/android/extension/AndroidExtensions.kt]: Okay @AjeshRPai 🙂  . I have moved the method `showSignInForResultBasedOnIsJetpackAppBuildConfig` from `ActivityLauncher`  to `WPMainActivity`. Done in commit 9da995b205af10f470739fbfe3920df5f47716fe

File: .idea/inspectionProfiles/Project_Default.xml
Comment (SowmiaS) [reply .idea/inspectionProfiles/Project_Default.xml]: Sorry, It was my mistake. I have removed the trailing newline from `inspectionProfiles/Project_Default.xml`. Done in commit d3cc7a6e55a3ca45751e8553a7eead1d47a9f752

**Comment by SowmiaS** [comment]
@AjeshRPai Thanks for taking your time to review my changes. I have fixed all the review comments given by you.

> In the meanwhile may I suggest that you update the PR by merging the [wordpress-mobile:trunk](https://github.com/wordpress-mobile/WordPress-Android)

I have merged the latest code from [wordpress-mobile:trunk](https://github.com/wordpress-mobile/WordPress-Android)


> As @AliSoftware Pointed out, Could you add a line to mention your fix in RELEASE-NOTES.txt as well?. You can find some examples in the same file.

As pointed out by @AliSoftware and you, I have mentioned the fix in RELEASE-NOTES.txt.

**Review by AjeshRPai — COMMENTED** [review]
I have looked at the code changes and tested it. Everything looks good to me. Great Job @SowmiaS . 👍 🙌 . Once again, Thank you for the contribution. I will merge this changes once the CI is done.

**Review by AjeshRPai — APPROVED** [review]

File: RELEASE-NOTES.txt
Comment (AjeshRPai) [inline RELEASE-NOTES.txt]: @SowmiaS Hi 👋🏼 , Could you move the line describing your fix to the milestone - 19.6. Am sorry for not communicating it earlier or updating the milestone tag in PR.

**Review by AjeshRPai — CHANGES_REQUESTED** [review]

File: RELEASE-NOTES.txt
Comment (SowmiaS) [reply RELEASE-NOTES.txt]: No problem at all @AjeshRPai 🙂. I have updated the RELEASE-NOTES.txt, moving the release note related to this PR from milestone 19.5 to 19.6. Done in 6b5f5e96d95bbe11ee36dd54da23c30e451ef067

**Review by AjeshRPai — APPROVED** [review]

**Comment by AliSoftware** [comment]
Good news @SowmiaS ! Your fix is now live and included in the new version (19.6) of WordPress Android that just shipped to the PlayStore 🎉 

Thanks again for your contribution!

**Pull request merged**
